Improve number formatting

jakubcolony opened the pull request:
## Description

This PR attempts to refactor number formatting in the dapp following the rules outlined in https://www.notion.so/Number-formatting-proposal-2eb2b390861349938b9e30d9abb583fb.

Looking forward to hearing your thoughts on the solution.

## Changes
- Added types for `move-decimal-point` library
- Refactored `Numeral` component:
  - It now follows the rules as described in the doc
  - It got simplified by removing the appearance prop (actually, most of the consuming components have been overriding it with own custom styles) and the ability to override the formatting rules (it wasn't used anywhere)
- Edited the consuming components to work with the new `Numeral` component

## Deletions
- Removed unused `PayoutsList` component

## TODO

- [x] Ensure consistency with spacing 
- [x] New Numeral component
- [x] Switch everything to use the new Numeral component

Resolves #3733

willm30 commented:
Don't forget to select "JoinColony/frontend" from the reviewers list so we get a notification when you're ready for a review 👍

willm30 reviewed (commented):
Nice work Jakub! I've been looking for inconsistencies with [the proposal](https://www.notion.so/colony/Number-formatting-proposal-2eb2b390861349938b9e30d9abb583fb) but only found one thing I wanted to query, which is the treatment of large billions. 

999,999,999,999 was rounded to 1T, and 999,000,000,000 was 0.999T, which I don't see in the proposal. I was expecting 999.999B and 999B respectively. 

Also, with respect to tiny numbers (0.00000...), the tooltip is not currently implemented on the action page, which I think it should be? I actually like the idea of engineering notation here, as at a glance I think it would be more informative than a string of zeroes. Then we could also have a tooltip as well, if the user wants to see the original value. I mention since you raised the question in the proposal, so I'd be interested to see how it looks. 

Final couple of things. There are a few build errors in your second commit re the omission of BigNumber type from the type definition of `moveDecimal`. 

And my instinct with regard to the removal of the appearance prop is that it should be left in and continued to be overriden where necessary. I say this only because I expect it was designed with that in mind originally, and follows the same custom as other core components in the dapp. But I might be mistaken here, so worth a second opinion.

jakubcolony commented on src/modules/core/components/Numeral/NewNumeral.tsx:
```diff
@@ -0,0 +1,120 @@
+  value: string,
+  { prefix, suffix }: PrefixSuffixOptions,
+) => {
+  return `${prefix ? `${prefix} ` : ''}${value}${suffix ? ` ${suffix}` : ''}`;
```
That's what I initially did as well, turns out we have a rule against it:
<img width="427" alt="image" src="https://user-images.githubusercontent.com/112586815/197070330-f0c41e18-8463-4f7e-a49c-da91536d13ef.png">
I'm not sure it's worth adding comments to disable this for a minor improvement (if any) in readability

Noted your point on removing the unnecessary brackets and return statement, thanks!

willm30 replied:
Fair enough 👍

jakubcolony commented on src/modules/core/components/Numeral/NewNumeral.tsx:
```diff
@@ -0,0 +1,120 @@
+import React from 'react';
```
Good spot!

jakubcolony commented on src/modules/core/components/Numeral/NewNumeral.tsx:
```diff
@@ -0,0 +1,120 @@
+  className?: string;
+}
+
+const NewNumeral = ({
```
Removed this file but I was also missing it in other components - fixed

jakubcolony commented:
>999,999,999,999 was rounded to 1T, and 999,000,000,000 was 0.999T, which I don't see in the proposal. I was expecting 999.999B and 999B respectively.

Nice catch! Should be fixed now

>Also, with respect to tiny numbers (0.00000...), the tooltip is not currently implemented on the action page, which I think it should be?

I have only updated the number formatting in places where the `Numeral` component had been already used and this doesn't seem to be the case for the action page. I'm wondering if there's a reason behind it, do you know?

>I actually like the idea of engineering notation here, as at a glance I think it would be more informative than a string of zeroes. Then we could also have a tooltip as well, if the user wants to see the original value. I mention since you raised the question in the proposal, so I'd be interested to see how it looks.

Thanks for weighing in, since both you and Arren have proposed this I will update the formatting to use engineering notation for tiny numbers.

>And my instinct with regard to the removal of the appearance prop is that it should be left in and continued to be overriden where necessary. I say this only because I expect it was designed with that in mind originally, and follows the same custom as other core components in the dapp.

My reason for removing it was that despite being specified, it was actually pointless in all but 2 or 3 cases as it was overwritten by consuming component's styles. But I see your point, I will bring the `appearance` prop back for these 2-3 cases I mentioned

jakubcolony commented:
Hey @willm30, I think I addressed all of your comments. 
As I mentioned, I brought back the `appearance` prop and it only seems relevant in one case. There was an interesting example where by setting theme to `grey`, it would only change the first letter colour, presumably to cater for this single use case in Create Payment Dialog:
<img width="96" alt="image" src="https://user-images.githubusercontent.com/112586815/197530166-3874480b-5092-4fa2-a4b4-f7baead500a2.png">
I find changing this via the `theme` prop confusing so I left it up to the consuming component to set such styles. Let me know if you disagree.

willm30 reviewed (commented):
I'm seeing the following with 999,999,999,999. Should be 999.999B?

![1000b](https://user-images.githubusercontent.com/64402732/197755028-f2edd92b-f17c-40f6-8911-ba9c8e072153.png)

Also not seeing the engineering notation for tiny numbers. Here I entered `0.00...1`. (can't tell you exactly as there's no way currently to see the original value). 

![lowz](https://user-images.githubusercontent.com/64402732/197756350-110b1d97-58bd-4ed0-9e26-7ce29bd6f945.png)

willm30 commented:
> I have only updated the number formatting in places where the `Numeral` component had been already used and this doesn't seem to be the case for the action page. I'm wondering if there's a reason behind it, do you know?

Hmm not sure. I think that would be a separate issue in any case so let's leave as is.

jakubcolony commented:
Closing this as it'll now be addressed in the CDapp [issue #53](https://github.com/JoinColony/colonyCDapp/issues/53)
